core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/ShardsLimitAllocationDecider.java

```diff
@@ -188,4 +159,9 @@ public Decision canAllocate(RoutingNode node, RoutingAllocation allocation) {
         return allocation.decision(Decision.YES, NAME, "the shard count is under node limit [%d] of total shards per node",
                 clusterShardLimit);
     }
+
+    @FunctionalInterface
+    private interface LimitPredicate<Integer>{
```

Comment:
I don't think that we need a new interface, we can just use `BiPredicate` (the only argument for a new interface is to avoid boxing by creating our own interface like `BiIntPredicate` that accepts two `int` arguments (as opposed to `Integer` arguments) but I don't think we need that here. Let's just stick with `BiPredicate`.